site/src/components/Resources/AppLink/AppLink.tsx

```diff
@@ -98,11 +103,46 @@ export const AppLink: FC<AppLinkProps> = ({ app, workspace, agent }) => {
             canClick
               ? (event) => {
                   event.preventDefault();
-                  window.open(
-                    href,
-                    Language.appTitle(appDisplayName, generateRandomString(12)),
-                    "width=900,height=600",
-                  );
+                  // This is an external URI like "vscode://", so
+                  // it needs to be opened with the browser protocol handler.
+                  if (app.external && !app.url.startsWith("http")) {
```

Comment:
I'm still pretty convinced this should be an `||`

Reply:
If `app.external` and the URL is `https://my-internal-jira.com` we wouldn't want to replace `window.location.href`, otherwise it would redirect the user from the dashboard.

Reply:
@aslilac lmk async if I'm thinking of this wrong, going to merge since it's blocking some registry work, but I'm happy to change it!

Reply:
OH. I think I misunderstood what `external` meant here. It's probably fine.